content/blog/2018-08-14-where-to-bird.md

```diff
@@ -0,0 +1,365 @@
+extracted with `osmdata`.
+
+``` r
+coords <- data.frame(lon = mpi$geometry.lng,
+                     lat = mpi$geometry.lat)
+crs <- sf::st_crs(results$osm_points)
+
+mpi_sf <- sf::st_as_sf(coords,
+                       coords = c("lon", "lat"), 
+                      crs = crs)
+
+points_map <- add_osm_objects(points_map, mpi_sf, 
+                             col = 'white',
+                             size = 5)
+```
+
+We can now visualize both points on the map, the MPI in white and the
+bird hide in salmon, South-West from the MPI.
+
+``` r
+points_map %>%
+  add_axes() %>%
+  print_osm_map (filename = 'map_a2.png', 
+                 width = 600,
+                 units = 'px', dpi = 72) 
+
+magick::image_read('map_a2.png') %>%
+  magick::image_annotate("Map data © OpenStreetMap contributors",
+                         color = "white",
+                         boxcolor =  "black",
+                         size = 15,
+                         gravity = "south")
+```
+
+<img src="/img/blog-images/2018-08-14-where-to-bird/map2-print-1.png" width="600" />
+
+Aha, now we see where the bird hide is, fantastic! But as Mark noted,
+birds can actually be observed from other places.
+
+### Birding where birds should be?
+
+Birds are most likely to be found **where water lies close to natural
+areas**, and we can translate this to R code! We shall get all water and
+(separately) all non-watery natural areas and find the shortest
+distances between them before plotting the results using
+`add_osm_surface`.
+
+First, let’s get all water and (separately) all non-watery natural
+areas.
+
+``` r
+dat <- opq(bbox = bbox) %>%
+     add_osm_feature(key = 'natural') %>%
+     osmdata_sf (quiet = FALSE)
+```
+
+    ## Issuing query to Overpass API ...
+
+    ## Rate limit: 2
+
+    ## Query complete!
+
+    ## converting OSM data to sf format
+
+``` r
+indx_W <- which (dat$osm_polygons$natural %in% c ("water", "wetland"))
+indx_N <- which (!dat$osm_polygons$natural %in% c ("water", "wetland"))
+
+xy_W <- sf::st_coordinates (dat$osm_polygons [indx_W, ])
+xy_N <- sf::st_coordinates (dat$osm_polygons [indx_N, ])
+```
+
+Then we use Mark’s [`geodist`
+package](https://github.com/hypertidy/geodist) to get pairwise distances
+between all points, find minima, and make a data.frame to submit to
+`add_osm_surface()`. We have 7068 watery points and 10065 non-watery
+points so the speed of `geodist` is crucial here!
+
+``` r
+t1 <- Sys.time()
+d <- geodist::geodist (xy_W, xy_N)
+# so fast!!!
+Sys.time() - t1
+```
+
+    ## Time difference of 13.57983 secs
+
+``` r
+d1 <- apply (d, 1, min)
+d2 <- apply (d, 2, min)
+xy <- cbind (rbind (xy_W, xy_N), c (d1, d2))
+xy <- xy [, c (1, 2, 5)]
+colnames (xy) <- c ("x", "y", "z")
+xy <- xy [!duplicated (xy), ]
+```
+
+Finally we plot the results on the roads we had gotten earlier, although
+we do not recommend staying on the middle of a road to bird! We re-add
```

Comment:
this phrase doesn't sound right: "..staying on the middle of a road to bird!"
"to see a bird"?

Reply:
changed